marlowe-integration-tests/test/Language/Marlowe/Runtime/Web/Common.hs

```diff
@@ -0,0 +1,69 @@
+module Language.Marlowe.Runtime.Web.Common
+  where
+
+import Cardano.Api
+  ( AsType(..)
+  , ShelleyWitnessSigningKey(..)
+  , TextEnvelope(..)
+  , TextEnvelopeType(..)
+  , deserialiseFromTextEnvelope
+  , serialiseToTextEnvelope
+  , signShelleyTransaction
+  )
+import Cardano.Api.SerialiseTextEnvelope (TextEnvelopeDescr(..))
+import Control.Concurrent (threadDelay)
+import Control.Monad.IO.Class (MonadIO(liftIO))
+
+import qualified Data.Set as Set
+import qualified Data.Text as T
+import qualified Language.Marlowe as V1
+import Language.Marlowe.Runtime.Integration.Common
+import Language.Marlowe.Runtime.Transaction.Api (WalletAddresses(..))
+import qualified Language.Marlowe.Runtime.Web as Web
+import Language.Marlowe.Runtime.Web.Client (getContract, postContract, putContract)
+import Language.Marlowe.Runtime.Web.Server.DTO (ToDTO(toDTO))
+import Servant.Client (ClientM)
+
+createCloseContract :: Wallet -> ClientM Web.TxOutRef
+createCloseContract Wallet{..}= do
+  let WalletAddresses{..} = addresses
+  let webChangeAddress = toDTO changeAddress
+  let webExtraAddresses = Set.map toDTO extraAddresses
+  let webCollataralUtxos = Set.map toDTO collateralUtxos
+
+  Web.CreateTxBody{txBody = createTxBody, ..} <- postContract
+    webChangeAddress
+    (Just webExtraAddresses)
+    (Just webCollataralUtxos)
+    Web.PostContractsRequest
+      { metadata = mempty
+      , version = Web.V1
+      , roles = Nothing
+      , contract = V1.Close
+      , minUTxODeposit = 2_000_000
+      , tags = mempty
+      }
+
+  createTx <- liftIO $ signShelleyTransaction' createTxBody signingKeys
+  putContract contractId createTx
+  _ <- waitUntilConfirmed (\Web.ContractState{status} -> status) $ getContract contractId
```

Comment:
Why is the result discarded here? Is it important that `getContract` performs its effect until its status is `Web.Confirmed`?